src/wepy/resampling/novelty/weight_novelty.py

```diff
@@ -0,0 +1,41 @@
+import numpy as np
+
+from wepy.rsampling.novelty.novelty import Novelty
+
+class Weight_Factor_Novelty(Novelty):
+
+    def __init__(self, n_walkers, pmin=1e-12):
+        
+        # The minimum weight for a walker
+        self.pmin = pmin
+        
+        # log(probability_min)
+        self.lpmin = np.log(pmin / 100)
+
+        # The total number of walkers
+        assert n_walkers is not None, "Must give the total number of walkers"
```

Comment:
Put this in the beginning to do validation before any business code is executed.